app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description(controller_name, action_name).to_json[:params]
```

Comment:
This will get API v2 definitions for regular controllers, so yes? We can add more documentation or override it in some controllers if necessary, for the general case I think it should work & forces us to write correct API documentation.